app/lib/screens/home/home.dart

```diff
@@ -1,10 +1,12 @@
-import 'package:app/navigation.dart';
+import 'package:app/navbar/navigation.dart';
+import 'package:app/sidebar/sidebar.dart';
 import 'package:flutter/material.dart';
 
 class HomePage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return Scaffold(
+      drawer: SideBar(),
```

Comment:
The build will run if state changes, but in this case it won't because it's stateless at the moment.  If you want state to persist, you need to use some kind of state management system. I included a todo inside main.dart for state management.

Reply:
Then I'm thinking we could create the generic drawer before any other pages are made and used, and give this generic drawer to those pages which have the menu option. I think the drawer isn't meant to be stateful?

Reply:
Correct, Drawer widget should be stateless (as you can see in the code).

Reply:
We're on the same page then.

- [ ] When you go to make stateful classes, could you investigate injecting the drawer just once (meaning, every time the widget/or whatever is updated, that it doesn't need the drawer every time.

Reply:
> We're on the same page then.
> 
> * [ ]  When you go to make stateful classes, could you investigate injecting the drawer just once (meaning, every time the widget/or whatever is updated, that it doesn't need the drawer every time)

I believe your concern can be resolve by using stack. I'm still new to flutter, but the stack I'm familiar is called IndexStack. I already implemented this in my bottom nav bar experimentation. I can explain this in further detail in our next meeting.